HACKING.md

```diff
@@ -0,0 +1,409 @@
+# Developing Salt
+
+
+So you want to contribute to the Salt project? Excellent! You can help in a
+number of ways:
+
+- Using Salt and opening bug reports.
```

Comment:
**Suggestion (non-blocking):** This is a long doc. Let's add a table of contents or an overview section that describes the different sections of the docs and links to their headers. I can help compose this if you need.

Reply:
Yeah, that's one reason .rst makes me grumpy :joy: 

With .md you can use anchors and relative links, and it's pretty unobtrusive. I don't even know what you'd do with .rst :stuck_out_tongue_closed_eyes: 

That being said, I'm strongly leaning towards yoinking the contents into the `docs/topics` somewhere and linking to that file from CONTRIBUTING, since if you get there from cloning the repo or you get there from the Internet that should be enough to get one started :thinking:

Reply:
If it can't be auto-generated (the ideal), maybe we should add an Overview section. I can write that if we want to spin off a ticket for it.